File: pinyougou/pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    //访问service的接口,然后自动调用实现类的方法
    //查询所有商品信息
    public List<TbBrand> findAll();
    //根据页码数查询页数
    public PageResult findPage(int pageNum,int pageSize);
    //增加
    public void add(TbBrand brand);
    //修改查询
    public TbBrand findOne(Long id);
    //修改之前查询的
    public void update(TbBrand brand);
    //删除品牌
    public void delete(Long[] ids);
    //条件查询
    public PageResult findPage(TbBrand brand,int pageNum,int pageSize);

    List<Map> selectOptionList();
}

File: pinyougou/pinyougou-parent/pinyougou-shop-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }
    this.findpage=function (page,rows) {
        return $http.get("../brand/findPage.do?page=" + page + "&rows=" + rows);
    }
    this.findOne=function (id) {
        return $http.post("../brand/findOne.do?id=" + id);
    }
    this.update=function (entity) {
        return $http.post("../brand/update.do", entity)
    }
    this.add=function (entity) {
        return $http.post("../brand/add.do", entity)
    }
    this.dele=function (ids) {
        return $http.get("../brand/delete.do?ids=" + ids)
    }
    this.search=function (page,rows,searchEntity) {
        return $http.post("../brand/search.do?page=" + page + "&rows=" + rows, searchEntity)
    }
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do')
    }

});